culturefeed_social/includes/blocks.inc

```diff
@@ -362,12 +362,24 @@ function culturefeed_social_user_list_block_for_activity($object, $activity_type
       break;
   }
 
-  try {
-    $activities = DrupalCultureFeed::searchActivities($query);
+  $cache_prefix = 'culturefeed:social:activities:';
+   
+  // If user isn't logged in try to get activities from cache. 
+  if (!culturefeed_is_culturefeed_user() && ($cache = cache_get($cache_prefix . $id . ':' . $activity_type, 'cache_culturefeed'))) {
+    $activities = $cache->data;
   }
-  catch (Exception $e) {
-    watchdog_exception('culturefeed_social', $e);
-    return;
+
+  else {
+    try {
+      $activities = DrupalCultureFeed::searchActivities($query);
```

Comment:
Ideally, the caching mechanism would be in the `DrupalCultureFeed::searchActivities` method, not here. The if-else here could also be dropped then. I see no need for checking if the user is logged in or not because the activities are always fetched (be it from cache if possible).

Reply:
I think the main concern for this was to be sure that users who perform a new activity (like, attend, recommend event,...) should always get the non-cached activities for an event. Otherwise they wouldn't see their own profile picture appearing after posting a new activity.

But indeed, this shouldn't be necessary for all logged in users. Maybe it would be a better approach to clear all caches for a certain event, only when it's updated by a new activity.

If this can be done in a better way, don't hesitate to move/change this.

Reply:
You should check the cache layer class of DrupalCulturefeed: https://github.com/cultuurnet/culturefeed/blob/master/lib/Drupal/DrupalCultureFeed_Cache.php#L126

Use the skipCache option for the loggedin users and you have the same result, or modify the code there in general. You are now fixing it for this block, but there are multiple blocks that show activities.

It should also not be needed. Every time an activity is performed or deleted, the whole cache of activities is cleared:
https://github.com/cultuurnet/culturefeed/blob/master/lib/Drupal/DrupalCultureFeed_Cache.php#L84
https://github.com/cultuurnet/culturefeed/blob/master/lib/Drupal/DrupalCultureFeed_Cache.php#L102
https://github.com/cultuurnet/culturefeed/blob/master/lib/Drupal/DrupalCultureFeed_Cache.php#L108